lib/nexpose/ajax.rb

```diff
@@ -158,14 +158,28 @@ def request(nsc, request, timeout = nil)
           raise Nexpose::AuthenticationFailed.new(response)
         else
           req_type = request.class.name.split('::').last.upcase
-          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
         end
       else
         req_type = request.class.name.split('::').last.upcase
-        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
       end
     end
 
+    def get_api_version(request)
```

Comment:
Would agree with Erran's comment - should make a difference between requesting an API version, versus getting the API version from a request.